fix: correct javadoc warnings

```
> Task :javadoc
/Users/andredietisheim/Documents/jboss-workspaces/intellij-common/src/main/java/com/redhat/devtools/intellij/common/utils/ConfigHelper.java:126: warning: no description for @param
     * @param thisContext
       ^
/Users/andredietisheim/Documents/jboss-workspaces/intellij-common/src/main/java/com/redhat/devtools/intellij/common/utils/ConfigHelper.java:127: warning: no description for @param
     * @param thatContext
       ^
/Users/andredietisheim/Documents/jboss-workspaces/intellij-common/src/main/java/com/redhat/devtools/intellij/common/utils/DownloadHelper.java:119: warning: no description for @throws
         * @throws IOException
           ^
/Users/andredietisheim/Documents/jboss-workspaces/intellij-common/src/main/java/com/redhat/devtools/intellij/common/utils/HighSensitivityRegistrar.java:43: warning: no @return
        protected boolean isMac() {
                          ^
/Users/andredietisheim/Documents/jboss-workspaces/intellij-common/src/main/java/com/redhat/devtools/intellij/common/utils/HighSensitivityRegistrar.java:55: warning: no description for @return
         * @return
           ^
/Users/andredietisheim/Documents/jboss-workspaces/intellij-common/src/main/java/com/redhat/devtools/intellij/common/utils/HighSensitivityRegistrar.java:79: warning: no @return
        protected Class<Modifier> getSensitivityWatchEventModifierClass() throws ClassNotFoundException {
                                  ^
/Users/andredietisheim/Documents/jboss-workspaces/intellij-common/src/main/java/com/redhat/devtools/intellij/common/utils/HighSensitivityRegistrar.java:79: warning: no @throws for java.lang.ClassNotFoundException
        protected Class<Modifier> getSensitivityWatchEventModifierClass() throws ClassNotFoundException {
                                  ^
/Users/andredietisheim/Documents/jboss-workspaces/intellij-common/src/main/java/com/redhat/devtools/intellij/common/tree/StructureTreeModelFactory.java:28: warning: no description for @throws
         * @throws IllegalAccessException
           ^
/Users/andredietisheim/Documents/jboss-workspaces/intellij-common/src/main/java/com/redhat/devtools/intellij/common/tree/StructureTreeModelFactory.java:29: warning: no description for @throws
         * @throws InvocationTargetException
           ^
/Users/andredietisheim/Documents/jboss-workspaces/intellij-common/src/main/java/com/redhat/devtools/intellij/common/tree/StructureTreeModelFactory.java:30: warning: no description for @throws
         * @throws InstantiationException
           ^
/Users/andredietisheim/Documents/jboss-workspaces/intellij-common/src/main/java/com/redhat/devtools/intellij/common/tree/StructureTreeModelFactory.java:31: warning: no description for @throws
         * @throws NoSuchMethodException
           ^
11 warnings
```

--- a/src/main/java/com/redhat/devtools/intellij/common/tree/StructureTreeModelFactory.java
+++ b/src/main/java/com/redhat/devtools/intellij/common/tree/StructureTreeModelFactory.java
@@ -25,10 +25,10 @@ public class StructureTreeModelFactory {
 	 * @param structure the structure to associate
 	 * @param project   the IJ project
 	 * @return the build model
-	 * @throws IllegalAccessException
-	 * @throws InvocationTargetException
-	 * @throws InstantiationException
-	 * @throws NoSuchMethodException
+	 * @throws IllegalAccessException if the model class could not be loaded
+	 * @throws InvocationTargetException if the model class could not be loaded
+	 * @throws InstantiationException if the model could not be instantiated
+	 * @throws NoSuchMethodException if the model class could not be loaded
 	 */
 	public static StructureTreeModel<AbstractTreeStructure> create(AbstractTreeStructure structure, Project project)
 			throws IllegalAccessException, InvocationTargetException, InstantiationException, NoSuchMethodException {

--- a/src/main/java/com/redhat/devtools/intellij/common/utils/ConfigHelper.java
+++ b/src/main/java/com/redhat/devtools/intellij/common/utils/ConfigHelper.java
@@ -123,8 +123,8 @@ public class ConfigHelper {
      *     <li>extensions</li>
      * </ul>
      *
-     * @param thisContext
-     * @param thatContext
+     * @param thisContext the fist context to compare
+     * @param thatContext the second context to compare
      * @return true if both contexts are equal
      *
      * @see NamedContext

--- a/src/main/java/com/redhat/devtools/intellij/common/utils/ConfigHelper.java
+++ b/src/main/java/com/redhat/devtools/intellij/common/utils/ConfigHelper.java
@@ -123,7 +123,7 @@ public class ConfigHelper {
      *     <li>extensions</li>
      * </ul>
      *
-     * @param thisContext the fist context to compare
+     * @param thisContext the first context to compare
      * @param thatContext the second context to compare
      * @return true if both contexts are equal
      *

--- a/src/main/java/com/redhat/devtools/intellij/common/utils/DownloadHelper.java
+++ b/src/main/java/com/redhat/devtools/intellij/common/utils/DownloadHelper.java
@@ -116,7 +116,7 @@ public class DownloadHelper {
          * @param toolName the name of the tool to download
          * @param url the URL to the tool description file
          * @return the command path
-         * @throws IOException
+         * @throws IOException if the tool was not found in the config file
          */
     public String downloadIfRequired(String toolName, URL url) throws IOException {
         ToolsConfig config = ConfigHelper.loadToolsConfig(url);

--- a/src/main/java/com/redhat/devtools/intellij/common/utils/HighSensitivityRegistrar.java
+++ b/src/main/java/com/redhat/devtools/intellij/common/utils/HighSensitivityRegistrar.java
@@ -38,7 +38,10 @@ public class HighSensitivityRegistrar {
 	}
 
 	/**
-	 * for testing purposes
+	 * Returns {@code true} if the current this is running on is MacOS. Returns {@code false} otherwise.
+	 * Method could be private but is kept protected for testing purposes.
+	 *
+	 * @return true if the current system is MacOS
 	 **/
 	protected boolean isMac() {
 		return SystemInfo.isMac;
@@ -52,7 +55,7 @@ public class HighSensitivityRegistrar {
 	 * Returns the SensitivityWatchEventModifier#HIGH if it exists (on sun jvms or
 	 * openjdk), {@code null} otherwise.
 	 *
-	 * @return
+	 * @return SensitivityWatchEventModifier#HIGH if it exists
 	 */
 	public Modifier get() {
 		try {
@@ -74,7 +77,11 @@ public class HighSensitivityRegistrar {
 	}
 
 	/**
-	 * for testing purposes
+	 * Returns the class SensitivityWatchEventModifier if it exists. Throws ClassNotFoundException if it was not found.
+	 * Method could be private, is kept protected for testing purposes.
+	 *
+	 * @return SensitivityWatchEventModifier if it was found
+	 * @throws ClassNotFoundException if the modifier class could not be found
 	 **/
 	protected Class<Modifier> getSensitivityWatchEventModifierClass() throws ClassNotFoundException {
 		return (Class<Modifier>) Class.forName(QUALIFIED_CLASSNAME);

